### ISSUE TITLE
feature: pass identifier to factory function

### ISSUE THREAD

**Issue by NoNameProvided** [typestack/typedi#171]
### Description
> Note: This issue is copied from PR https://github.com/typestack/typedi/pull/82.

Pass the service identifier to the factory functions, this is useful for factory functions that handle the creation of multiple types of services.

### Proposed solution

Without identifier:
```ts
@Service({
  factory: (container: ContainerInstance) => {
    const manager = container.get(EntityManager)
    return manager.getCustomRepository(AccountRepository)
  },
})
@EntityRepository(Account)
export class AccountRepository extends BaseRepository<Account> { }
```

With identifier:
```ts
// This factory can now be made generic rather than having to be re-defined for every single repository.
@Service({
  factory: (container: ContainerInstance, identifier: any) => {
    const manager = container.get(EntityManager)
    return manager.getCustomRepository(identifier)
  },
})
@EntityRepository(Account)
export class AccountRepository extends BaseRepository<Account> {
```

**Comment by NoNameProvided** [comment]
Added in 565b9e910f5ce7352f853ce6ea5ecaccf7932955.

**Comment by NoNameProvided** [comment]
Released as `0.10.0`.
